### PR TITLE
test: add test for custom stacks

### PR DESCRIPTION
**Pull request by jhockett**
Test custom stack flow with SES

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by kaustavghosh06** [comment]
@yuth could you please glance at this PR as well?

**Comment by jhockett** [comment]
> Why do we need a separate step in CircleCI for this?

I'm pretty sure the CircleCI changes were generated since this is a new file in the ```__tests__``` folder.

File: .circleci/config.yml [hunk 1]
```diff
@@ -982,14 +990,18 @@ workflows:
...
-      - plugin-amplify_e2e_tests:
+      - customStacks-amplify_e2e_tests:
           filters: &ref_2
             branches:
               only:
                 - master
                 - split-test
           requires:
             - publish_to_local_registry
+      - plugin-amplify_e2e_tests:
```
Comment (kaustavghosh06) [inline .circleci/config.yml:1001]: Why do have this? Isn't this unrelated?

**Comment by yuth** [comment]
> Why do we need a separate step in CircleCI for this?



> > Why do we need a separate step in CircleCI for this?
> 
> I'm pretty sure the CircleCI changes were generated since this is a new file in the `__tests__` folder.

Yes. This is done to split tests into separate job in CircleCI

File: .circleci/config.yml (hunk 1 above)
Comment (yuth) [reply .circleci/config.yml:1001]: This is diffing algorithm doing its thing

File: packages/amplify-e2e-tests/src/__tests__/customStacks.test.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+    const meta = getProjectMeta(projRoot);
+
+    const { DeploymentBucketName: bucketName, Region: region } = meta.providers.awscloudformation;
+    expect(bucketName).toBeDefined();
+    expect(region).toBeDefined();
+    const bucketExists = await checkIfBucketExists(bucketName, region);
+    expect(bucketExists).toMatchObject({});
+
+    const { customResource } = meta.customCategory;
```
Comment (kaustavghosh06) [inline packages/amplify-e2e-tests/src/__tests__/customStacks.test.ts:110]: Out here - I beleive either you should make an SDK call to SES to check if the AWS resources exists or you should have an "Output" in the Cloudformation file which gets populated in the amplify-meta file after deployment which you can check for.

File: packages/amplify-e2e-tests/src/__tests__/customStacks.test.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  amplifyPushUpdate,
+  checkIfBucketExists,
+  createNewProjectDir,
+  deleteProject,
+  deleteProjectDir,
+  getProjectMeta,
+  initJSProjectWithProfile,
+} from 'amplify-e2e-core';
+import * as path from 'path';
+import { existsSync, mkdirSync, writeFileSync } from 'fs';
```
Comment (yuth) [inline packages/amplify-e2e-tests/src/__tests__/customStacks.test.ts:11]: we usually use `fs-extra`

**Pull request closed without merging**